core/primitives/src/views.rs

```diff
@@ -1653,6 +1653,26 @@ impl ExecutionOutcomeWithIdView {
     }
 }
 
+pub struct TxStatusView {
+    pub execution_outcome: Option<FinalExecutionOutcomeViewEnum>,
+    pub status: TxExecutionStatus,
```

Comment:
Oh, this is returned in the response? Interesting, I thought it will be part of the request, where the user specifies which finality guarantees they need.

Both approaches seem possible. From a user perspective, I imagine it is easier to ask for a specific finality and wait for the answer, as otherwise they have to keep polling until the desired finality is reached.
On the other hand, if we include the enum in the request, we have to do the polling on the RPC node which means more work on the server.

Reply:
I was also thinking about which approach is better.
I decided to move forward with returning status as fast as possible, with the finality info in the response.
And, when I'll work on `broadcast_tx`, I have a plan to add the same struct as an optional argument to the request.

Reply:
I think "optional for the request, always included in the response" makes sense. As long as a user asking for `Inclusion` only gets a response where `Inclusion` or strictly stronger finality is guaranteed, and it is well documented, I think it should be okay.

Reply:
Oh and yes, starting with just the field in the response first makes sense to me.